app/Listeners/GitHubStatusListener.php

```diff
@@ -20,27 +20,40 @@ class GitHubStatusListener {
    * @return void
    */
   public function handle(BuildCompletedEvent $event) {
-    $description = 'Build size: ' . Format::fileSize($event->total_size);
     $state = 'success';
-
     if ($event->has_base_build) {
-      // Can compare to base
-      $diff = $event->base_total_size - $event->total_size;
-      if (abs($diff) === 0) {
-       $description .= ' (no change)';
+
+      $largestChange = null;
+      $largestDiff = 0;
+      foreach($event->build_artifacts as $artifact){
+        foreach($event->base_build_artifacts as $base_artifact) {
```

Comment:
This is `O(n^2)` which is not ideal, and it should use the `name` of the ProjectArtifact rather than the `filename` of the BuildArtifact (as the BuildArtifact might have a version number in it).

Having said that, ... `base_build_artifacts` is already keyed by the `project_artifact_id` so you should be able to just use that:

```
foreach ($event->build_artifacts as $artifact) {
  if (!$event->base_build_artifacts->has($artifact->project_artifact_id) {
    // No base version, so nothing to compare to
    continue;
  }
  $base_artifact = $event->base_build_artifacts->get($artifact->project_artifact_id);
  $diff = $base_artifact->size - $artifact->size;
  ......
}
```

Look at the code that `GitHubCommentListener.php` is using for example :)

Reply:
While I understand your fears for using an algo that's O(n^2) I don't think the inputs size is ever going to be more than 100, therefore the impact is negligible.

That side I think your proposed solution is still better anyway, will update it :D